packages/react/src/reactrouterv6.tsx

```diff
@@ -141,6 +142,29 @@ function stripBasenameFromPathname(pathname: string, basename: string): string {
   return pathname.slice(startIndex) || '/';
 }
 
+function sendIndexPath(pathBuilder: string, pathname: string, basename: string): [string, TransactionSource] {
+  const reconstructedPath = pathBuilder || _stripBasename ? stripBasenameFromPathname(pathname, basename) : pathname;
+
+  const formattedPath =
+    // If the path ends with a slash, remove it
+    reconstructedPath[reconstructedPath.length - 1] === '/'
+      ? reconstructedPath.slice(0, -1)
+      : // If the path ends with a wildcard, remove it
+        reconstructedPath.slice(-2) === '/*'
+        ? reconstructedPath.slice(0, -1)
+        : reconstructedPath;
```

Comment:
I wonder if a regex replace is less expensive here. Would be interesting to benchmark.

Reply:
Just tried now, looks like it's slower in this case https://jsperf.app/gemolu